File: py3dpolys_le/_logging.py
import logging

_loggers = {}


def get_logger(name="py3dpolys_le"):
    global _loggers

    if name not in _loggers:
        _loggers[name] = logging.getLogger(name)
        _loggers[name].addHandler(logging.NullHandler())

    return _loggers[name]

File: src/Logging.jl
"""
Logging utilities for 3DPolyS-LE
"""
module LoggingUtils

# Import from Base.CoreLogging to avoid conflicts
import Base.CoreLogging: AbstractLogger, @info, @warn, @error, @debug, SimpleLogger, Info

const _loggers = Dict{String, AbstractLogger}()

function get_logger(name::String = "j3DPolySLE")
    global _loggers
    
    if !haskey(_loggers, name)
        _loggers[name] = SimpleLogger(stdout, Info)
    end
    
    return _loggers[name]
end

end # module

